### PR TITLE
Merge changes to move server into its own package

### PR DESCRIPTION
**Pull request by akalinin-stripe**
## Notify
r? @richardm-stripe 
cc @dcr-stripe @stripe/api-libraries

## Summary
Based on the changes from https://github.com/stripe/stripe-mock/pull/238 (thank you @Splizard) with conflicts resolved against the master branch.

The original intent was to fix issue #206 and allow running embedded version of the server.
Detailed changes:
- Move server code from main package into /server directory
- Split embedded data generation to embed spec in server directory and certificates only in main directory
- Refactor out global state `verbose` variable and include it into server, so it does not become part of the API (not sure what to do with `Version` though, usually it is OK to leave it as it is set by ldflags).

Finally, it looks like it's a lot of changes, but mostly it's shuffling code around to a different package.

## Testing
- Unit tests updated
- Ran tests with the current stripe-go repo

### CODE REVIEW

**Comment by akalinin-stripe** [comment]
False positive in golint is catching auto-generated file comment:
`sh
go list ./... | xargs -I{} -n1 sh -c 'golint -set_exit_status {} || exit 255'
/home/travis/gopath/src/github.com/stripe/stripe-mock/server/bindata.go:1:1: package comment should be of the form "Package server ..."
`

File: Makefile [hunk 1]
```diff
@@ -9,10 +9,8 @@ build:
 check-gofmt:
 	scripts/check_gofmt.sh
 
-# This command is overcomplicated because Golint's `./...` doesn't filter
-# `vendor/` (unlike every other Go command).
 lint:
-	go list ./... | xargs -I{} -n1 sh -c 'golint -set_exit_status {} || exit 255'
+	golint
```
Comment (akalinin-stripe) [inline Makefile:13]: golint is deprecated, and there's not much we can do to customize it's behavior. We have another task to replace it with a supported linter.

Comment (richardm-stripe) [reply Makefile:13]: https://github.com/stripe/stripe-go/pull/1288 is what we did for stripe-go

Comment (akalinin-stripe) [reply Makefile]: Updated to be the same in this repo.

File: main.go [hunk 2]
```diff
@@ -1,21 +1,18 @@
-//go:generate go-bindata -mode 444 -modtime 1 cert/cert.pem cert/key.pem openapi/openapi/fixtures3.json openapi/openapi/spec3.json
+//go:generate go-bindata -mode 0444 -modtime 1 cert/cert.pem cert/key.pem
```
Comment (akalinin-stripe) [inline main.go:1]: Caught by staticcheck - permissions need to be in octal format.

File: main.go [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
+//go:generate go-bindata -o server/bindata.go -pkg server -mode 0444 -modtime 1 openapi/openapi/fixtures3.json openapi/openapi/spec3.json
```
Comment (richardm-stripe) [inline main.go:2]: does this mean we should delete `bindata.go` at the root? `server/bindata.go` is the new path?

File: server/server.go [hunk 4]
```diff
@@ -133,13 +140,90 @@ type ResponseError struct {
...
 
+// LoadFixtures load fixtures from a JSON file
+//
+// If path is empty, fixtures are loaded from internal embedded assets.
+func LoadFixtures(fixturesPath string) (*spec.Fixtures, error) {
+	var data []byte
+	var err error
+
+	if fixturesPath == "" {
+		// And do the same for fixtures
```
Comment (richardm-stripe) [inline server/server.go:151]:
```suggestion
```

File: server/server.go [hunk 5]
```diff
@@ -133,13 +140,90 @@ type ResponseError struct {
...
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("error loading fixtures: %v", err)
+	}
+
+	var fixtures spec.Fixtures
+	err = json.Unmarshal(data, &fixtures)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding spec: %v", err)
```
Comment (richardm-stripe) [inline server/server.go:168]:
```suggestion
		return nil, fmt.Errorf("error decoding fixtures: %v", err)
```

**Review by richardm-stripe — CHANGES_REQUESTED** [review]
This all looks good! The only thing is that I think we should remove `bindata.go` as the path has moved.

**Review by richardm-stripe — APPROVED** [review]
Clarified IRL! I understand the bindata commands now! Good to merge!

File: main.go (hunk 3 above)
Comment (akalinin-stripe) [reply main.go:2]: Saving the comment for history here... The reason there are 2 bindata files is one of them is for fixtures/spec the other one is for the certs:
- server dir - includes fixtures3.json and spec3.json
- main dir - includes cert.pem and key.pem

**Pull request merged**